source/src/storage/cs_IpcRamBluenet.cpp

```diff
@@ -0,0 +1,120 @@
+/*
+ * Author: Crownstone Team
+ * Copyright: Crownstone (https://crownstone.rocks)
+ * Date: Oct 24, 2022
+ * License: LGPLv3+, Apache License 2.0, and/or MIT (triple-licensed)
+ */
+
+#include <ipc/cs_IpcRamData.h>
+#include <logging/cs_Logger.h>
+#include <storage/cs_IpcRamBluenet.h>
+
+#define LogIpcRamBluenetInfo LOGi
+#define LogIpcRamBluenetDebug LOGd
+#define LogIpcRamBluenetVerbose LOGvv
+#define LogLevelIpcRamBluenetVerbose SERIAL_VERY_VERBOSE
+
+IpcRamBluenet::IpcRamBluenet() {}
+
+IpcRamBluenet& IpcRamBluenet::getInstance() {
+	// Use static variant of singleton, no dynamic memory allocation
+	static IpcRamBluenet instance;
+	return instance;
+}
+
+void IpcRamBluenet::init() {
+	// Get bluenet data from IPC ram.
+	uint8_t ipcDataSize = 0;
+
+	IpcRetCode ipcCode  = getRamData(IPC_INDEX_BLUENET_TO_BLUENET, _ipcData.raw, &ipcDataSize, sizeof(_ipcData.raw));
+	if (ipcCode != IPC_RET_SUCCESS) {
+		LogIpcRamBluenetDebug("Failed to get IPC data: ipcCode=%i", ipcCode);
+		clearData();
+		return;
+	}
+	if (_ipcData.bluenetRebootData.ipcDataMajor != BLUENET_IPC_BLUENET_REBOOT_DATA_MAJOR) {
+		LogIpcRamBluenetInfo(
+				"Different major version: major=%u expected=%u",
+				_ipcData.bluenetRebootData.ipcDataMajor,
+				BLUENET_IPC_BLUENET_REBOOT_DATA_MAJOR);
+		clearData();
+		return;
+	}
+
+	// We need to ignore this warning, it's triggered because the minimum minor is 0, making the statement always false.
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wtype-limits"
+	if (_ipcData.bluenetRebootData.ipcDataMinor < BLUENET_IPC_BLUENET_REBOOT_DATA_MINOR) {
+		LogIpcRamBluenetInfo(
```

Comment:
and here too.
maybe set BLUENET_IPC_BLUENET_REBOOT_DATA_MINOR to 1 instead of pragma ignore error